File: test/fixtures/transforms/js-createandpoll-with-inline-create/input.js
import OpenAI from "openai";

const openai = new OpenAI();

export async function createReviewAssistant() {
  const assistant = await openai.beta.assistants.create({
    model: "gpt-4-turbo",
    instructions: "Review pull requests and flag risky changes.",
    tools: [{ type: "file_search" }],
    tool_resources: {
      file_search: { vector_store_ids: ["vs_review_docs"] },
    },
  });
  return assistant.id;
}

export async function askAssistant(assistantId, threadId, question) {
  await openai.beta.threads.messages.create(threadId, {
    role: "user",
    content: question,
  });
  const run = await openai.beta.threads.runs.createAndPoll(threadId, {
    assistant_id: assistantId,
  });
  const messages = await openai.beta.threads.messages.list(threadId);
  return messages.data[0].content[0].text.value;
}

File: test/fixtures/transforms/python-async-dance/input.py
import os

from openai import AsyncOpenAI

client = AsyncOpenAI()

ASSISTANT_ID = os.environ["OPENAI_ASSISTANT_ID"]


async def ask_assistant(prompt: str) -> str:
    thread = await client.beta.threads.create()
    await client.beta.threads.messages.create(
        thread_id=thread.id, role="user", content=prompt
    )
    run = await client.beta.threads.runs.create_and_poll(
        thread_id=thread.id, assistant_id=ASSISTANT_ID
    )
    messages = await client.beta.threads.messages.list(thread_id=thread.id)
    return messages.data[0].content[0].text.value
